src/app/space/create/deployments/services/deployments.service.ts

```diff
@@ -240,18 +239,17 @@ export class DeploymentsService implements OnDestroy {
       .distinctUntilChanged(deepEqual);
   }
 
-  getEnvironments(spaceId: string): Observable<ModelEnvironment[]> {
+  getEnvironments(spaceId: string): Observable<string[]> {
     // Note: Sorting and filtering out test should ideally be moved to the backend
     return this.getEnvironmentsResponse(spaceId)
       .map((envs: EnvironmentStat[]) => envs || [])
       .map((envs: EnvironmentStat[]) => envs.map((env: EnvironmentStat) => env.attributes))
       .map((envs: EnvironmentAttributes[]) => envs.sort((a, b) => -1 * a.name.localeCompare(b.name)))
       .map((envs: EnvironmentAttributes[]) => envs
         .filter((env: EnvironmentAttributes) => env.name !== 'test')
-        .map((env: EnvironmentAttributes) => ({ name: env.name } as ModelEnvironment))
+        .map((env: EnvironmentAttributes) => env.name)
       )
-      .distinctUntilChanged((p: ModelEnvironment[], q: ModelEnvironment[]) =>
-        deepEqual(new Set<string>(p.map(v => v.name)), new Set<string>(q.map(v => v.name))));
+      .distinctUntilChanged((p: string[], q: string[]) => deepEqual(new Set<string>(p), new Set<string>(q)));
```

Comment:
Can you test if this custom equality/comparison function used for `distinctUntilChanged` is still needed? The original reason it was included rather than using the default was because arrays of Environment objects did not compare as equal as expected (ie. [{ name: 'foo' }] != [{ name: 'foo' }]). Now that they're just string arrays I suspect it'll work as expected by just doing `.distinctUntilChanged()` with no comparison function argument.